### PR TITLE
Update EIP-20: Renovate

### PR DESCRIPTION
**Pull request by Pandapip1**
WIP. Likely to be contentious.

### CODE REVIEW

**Comment by lightclient** [comment]
Strongly against. We've never rewritten EIPs to fit the "new recommendations" and I don't think we should start.

**Comment by Pandapip1** [comment]
> We've never rewritten EIPs to fit the "new recommendations" and I don't think we should start.

We have *modified* EIPs to fit new recommendations before: https://github.com/ethereum/EIPs/pull/5055

**Comment by SamWilsn** [comment]
After today's EIPIP, I think the consensus is that this PR changes more than the absolute minimum to comply with the current EIP-1 guidelines. For example, moving Simple Summary to `description:` is probably fine, while changing the Solidity version is too much

**Comment by Pandapip1** [comment]
That seems reasonable. I'll revert the extra changes.

**Comment by Pandapip1** [comment]
@SamWilsn mind re-reviewing?

**Comment by Pandapip1** [comment]
@eth-bot rerun

File: EIPS/eip-20.md [hunk 1]
```diff
@@ -167,27 +142,42 @@ MUST trigger on any successful call to `approve(address _spender, uint256 _value
...
+2. Later, Alice decides to set the approval from $N$ to $M$ ( $M>0$ ), so she calls `approve(Bob, M)`
+3. Bob notices Alice's second transaction when it is submitted to the mempool
+4. Before Alice's `approve(Bob, M)` transaction is included in a block, Bob sends a transaction calling `transferFrom(Alice, Bob, N)` with a higer priority fee than Alice's transaction, and a transaction calling `transferFrom(Alice, Bob, M)` with a lower priority fee than Alice's transaction
+5. Bob's `transferFrom(Alice, Bob, N)` will be executed first, transfering $N$ tokens from Alice to Bob
+6. Next, Alice's `approve(Bob, M)` will execute, allowing Bob to transfer an additonal $M$ tokens
+7. Finally, Bob's `transferFrom(Alice, Bob, M)` will be executed first, transfering another $M$ tokens from Alice to Bob
 
+Bob was able to transfer a total of $N+M$ tokens instead of a total of $M$.
+
+(Written by Mikhail Vladimirov <mikhail.vladimirov@gmail.com> and Dmitry Khovratovich <khovratovich@gmail.com>, edited by Pandapip1 (@Pandapip1))
```
Comment (SamWilsn) [inline EIPS/eip-20.md:175]:
```suggestion
```

Generally we don't include attribution beyond the author field.

File: EIPS/eip-20.md [hunk 2]
```diff
@@ -1,52 +1,46 @@
...
-A standard interface for tokens.
-
-
 ## Abstract
 
-The following standard allows for the implementation of a standard API for tokens within smart contracts.
-This standard provides basic functionality to transfer tokens, as well as allow tokens to be approved so they can be spent by another on-chain third party.
-
+This EIP standardizes an interface for fungible tokens within smart contracts.
+The interface has basic transfer functionality, and allows tokens to be approved so they can be spent by another on-chain third party.
```
Comment (SamWilsn) [inline EIPS/eip-20.md:16]: I believe this change goes beyond the minimum required to fit current standards.

```suggestion
The following standard allows for the implementation of a standard API for tokens within smart contracts.
This standard provides basic functionality to transfer tokens, as well as allow tokens to be approved so they can be spent by another on-chain third party.
```

File: EIPS/eip-20.md [hunk 3]
```diff
@@ -119,21 +104,16 @@ The function SHOULD `throw` unless the `_from` account has deliberately authoriz
...
-
-
 #### approve
 
 Allows `_spender` to withdraw from your account multiple times, up to the `_value` amount. If this function is called again it overwrites the current allowance with `_value`.
 
-**NOTE**: To prevent attack vectors like the one [described here](https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/) and discussed [here](https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729),
-clients SHOULD make sure to create user interfaces in such a way that they set the allowance first to `0` before setting it to another value for the same spender.
-THOUGH The contract itself shouldn't enforce it, to allow backwards compatibility with contracts deployed before
+**NOTE**: Please read the [Security Considerations](#security-considerations) section for potential attack vectors.
```
Comment (SamWilsn) [inline EIPS/eip-20.md:111]: Seems like you're removing part of the specification?

```suggestion
clients SHOULD make sure to create user interfaces in such a way that they set the allowance first to `0` before setting it to another value for the same spender.
THOUGH The contract itself shouldn't enforce it, to allow backwards compatibility with contracts deployed before

**NOTE**: Please read the [Security Considerations](#security-considerations) section for potential attack vectors.
```

File: EIPS/eip-20.md [hunk 4]
```diff
@@ -167,27 +142,42 @@ MUST trigger on any successful call to `approve(address _spender, uint256 _value
...
+### The approve and transferFrom flow
 
+`approve` and `transferFrom` can be used in conjunction with one another to allow smart contracts to take an action if and only if a transfer succeeds. This enables use cases such as decentralized exchanges, which can trustlessly swap one token for another.
 
-## Implementation
+### Decimals is optional
 
-There are already plenty of ERC20-compliant tokens deployed on the Ethereum network.
-Different implementations have been written by various teams that have different trade-offs: from gas saving to improved security.
+Some tokens might wish to use a base unit other than $10^{-18}$, and sometimes it doesn't make sense to have a fraction of a token. However, not every token might reasonably need to specify `decimals`. Thus, `decimals` is optional.
```
Comment (SamWilsn) [inline EIPS/eip-20.md:153]: I don't think we should invent content that wasn't there in the original proposal.

File: EIPS/eip-20.md (hunk 3 above)
Comment (Pandapip1) [reply EIPS/eip-20.md:111]: I didn't think so. I'll readd it if it means this can be merged.

File: EIPS/eip-20.md (hunk 4 above)
Comment (Pandapip1) [reply EIPS/eip-20.md:156]: It's a non-normative required section. If I were the person writing ERC-20 and decided to include this field, that would be why. If the original authors consent/agree with the motivation provided then I see nothing wrong with this.

**Comment by Pandapip1** [comment]
Hey @SamWilsn, mind checking out the changes now?

File: EIPS/eip-20.md (hunk 4 above)
Comment (SamWilsn) [reply EIPS/eip-20.md:156]: This is still more than is required to comply with current guidelines, and I don't think this qualifies as errata.

**Comment by Pandapip1** [comment]
Oh lol that's a funny eip-review-bot bug. I'll get that fixed.

**Pull request closed without merging**